src/Uno.UWP/Extensions/PermissionHelper.Android.cs

```diff
@@ -0,0 +1,198 @@
+﻿#if __ANDROID__
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Windows.Extensions
+{
+	public partial class PermissionsHelper
+	{
+		private static TaskCompletionSource<bool> _permissionCompletionSource;
+
+		/// <summary>
+		/// Return null on error, or array of permission to be asked for (not granted at this time). Both parameters can be null
+		/// </summary>
+		/// <param name="requiredPermissions">Permissions that are required. Can be null.</param>
+		/// <param name="optionalPermissions">Permissions that are required only if declared in Manifest. Can be null.</param>
+		/// <returns>Array of all permission that is not granted at this moment (can be empty!), or null if some error occured</returns>
+		public static string[] MissingPermissions(string[] requiredPermissions, string[] optionalPermissions)
+		{
+			// since API 29, we should do something more:
+			// https://developer.android.com/reference/android/content/pm/PackageInstaller.SessionParams.html#setWhitelistedRestrictedPermissions(java.util.Set%3Cjava.lang.String%3E)
+
+			// do we have declared permissions in Manifest?
+			Android.Content.Context context = Android.App.Application.Context;
+			Android.Content.PM.PackageInfo packageInfo =
+				context.PackageManager.GetPackageInfo(context.PackageName, Android.Content.PM.PackageInfoFlags.Permissions);
+			var manifestPermissions = packageInfo?.RequestedPermissions;
+			if (manifestPermissions is null)
+				return null;
+
+			// test required permissions
+			if (requiredPermissions != null)
+			{
+				foreach (string permission in requiredPermissions)
+				{
+					bool foundInManifest = false;
+					foreach (string oPerm in manifestPermissions)
+					{
+						if (oPerm.Equals(permission, StringComparison.OrdinalIgnoreCase))
+							foundInManifest = true;
+
+					}
+					if (!foundInManifest) return null;

```

Comment:
Use braces, but in general, it would be best if you used Linq instead:

```csharp
if(!manifestPermissions.Any(p => perm.Equals(p, StringComparison.OrdinalIgnoreCase)))
{
   return null;
}
```

Same for the loops below.

Reply:
Are you sure that `manifestPermissions.Any(p => perm.Equals(p, StringComparison.OrdinalIgnoreCase))` is more readable than version without LINQ? :)